### ISSUE TITLE
Can you update the angularjs2 version from RC4 to RC5

### ISSUE THREAD

**Issue by sustwater** [akserg/ng2-dnd#57]
ng2-dnd is very good !!!

Can you update the angularjs2 version from RC4 to RC5 ?

**Comment by akserg** [comment]
Hi @sustwater 
I will do it very soon.

**Comment by pdelorme** [comment]
Hi

I get **ng2-dnd** working in RC.5 without change. 
However I get blocked when DND_DIRECTIVES is used in the declaration of two diferent @NgModule. 
I'm getting the folowing error message : 

> Type DraggableComponent is part of the declarations of 2 modules: SourceModule and TargetModule

This seem to be cause by DND not being a shared Module (see [https://github.com/angular/angular/issues/10646](url))

Any estimated release date for RC5 version ?

**Comment by pdelorme** [comment]
Here is the shared module I did to have ng2-dnd working in RC.5 in NgModule :

```
// dnd.shared.module.ts
import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';
import {DND_DIRECTIVES, DragDropService} from "ng2-dnd/ng2-dnd";

@NgModule({
    imports:      [ CommonModule ],
    declarations: [ DND_DIRECTIVES ],
    exports:      [ DND_DIRECTIVES ],
    providers: [DragDropService]
})
export class DndSharedModule {
    static forRoot(): ModuleWithProviders {
        return {
            ngModule: DndSharedModule
        };
    }
}
```
working like a charm. 
Hope it helps.

**Comment by sustwater** [comment]
Hi @akserg 
Thank you for your update.

There is a problem in my project.
error：Uncaught TypeError: Cannot read property 'type' of null
            [WDS] Errors while compiling.

my module is ：
`import { NgModule }      from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { HttpModule } from '@angular/http';  //http服务
import {DND_PROVIDERS, DND_DIRECTIVES} from 'ng2-dnd';

@NgModule({
  imports:      [ BrowserModule ,HttpModule ,DND_DIRECTIVES],
  declarations: [ HomepageComponent ],
  providers: [
    DND_PROVIDERS
  ],
  bootstrap:    [ HomepageComponent ]
})
export class MainModule { }`


Now has the latest example?

**Comment by hstevensagain** [comment]
@sustwater Does putting DND_DIRECTIVES inside the declarations array work?

**Comment by akserg** [comment]
Hi @sustwater. @hstevensagain is right. Please look at the example on README:

```
import {Component} from '@angular/core';
import {DND_PROVIDERS, DND_DIRECTIVES} from 'ng2-dnd';

@Component({
    selector: 'app',
    directives: [DND_DIRECTIVES],
    providers: [DND_PROVIDERS]
    template: `
```

Don't use directives inside `import` property.

**Comment by sustwater** [comment]
Hi @akserg  @hstevensagain  . Thank you very much!

**Comment by pdelorme** [comment]
I agrea with @daniele-zurico. typical DnD usage is between module, so shared modules should be part of standard release.

**Comment by Gillardo** [comment]
any news on this??

**Comment by theunreal** [comment]
RC6 is up and no longer works with this package

**Comment by choucry13** [comment]
@theunreal , i used ng2 like a shared-module with a rc6 project and it's work fine :-)

**Comment by Gillardo** [comment]
I am also having the same issues....

**Comment by cternes** [comment]
Any news on this? Angular 2 final is out but seems to be not working with ng2-dnd.

**Comment by akserg** [comment]
Fixed in 2.0.0

**Comment by avantika3667** [comment]
hi 
I am using angular2+webpack+es6

in package.json
 "angular2": "2.0.0-beta.9",

index.js
require('!!script!ng2-dnd/bundles/ng2-dnd.min.js');

and its showing error
Uncaught TypeError: core_1.NgModule is not a function
